src/move/econia/sources/core.move

```diff
@@ -223,99 +227,61 @@ module econia::core {
+        move_to(&object::generate_signer(&constructor_ref), OpenOrders {
             market_id,
             trading_pair,
             market_address,
             user: user_address,
-            market_account_address,
-            extend_ref,
-            base_balances: no_balances,
-            quote_balances: no_balances,
+            open_orders_address,
         });
         smart_table::add(
-            market_accounts_map_ref_mut,
-            trading_pair,
-            MarketAccountMetadata {
-                market_id,
-                trading_pair,
-                market_address,
-                user: user_address,
-                market_account_address,
-            }
-        );
-    }
-
-    public entry fun ensure_integrator_fee_store_registered(
-        integrator: &signer,
-        base_metadata: Object<Metadata>,
-        quote_metadata: Object<Metadata>,
-    ) acquires IntegratorFeeStores, Registry {
-        let integrator_address = signer::address_of(integrator);
-        if (!exists<IntegratorFeeStores>(integrator_address)) {
-            move_to(integrator, IntegratorFeeStores { map: smart_table::new() });
-        };
-        let integrator_fee_stores_map_ref_mut =
-            &mut borrow_global_mut<IntegratorFeeStores>(integrator_address).map;
-        let trading_pair = TradingPair { base_metadata, quote_metadata };
-        if (smart_table::contains(integrator_fee_stores_map_ref_mut, trading_pair)) return;
-        ensure_market_registered(integrator, base_metadata, quote_metadata);
-        let (market_id, market_address) =
-            get_market_id_and_address_for_registered_pair(trading_pair);
-        let (constructor_ref, extend_ref) =
-            create_nontransferable_sticky_object(integrator_address);
-        move_to(&object::generate_signer(&constructor_ref), IntegratorFeeStore {
+            open_orders_map_ref_mut,
             market_id,
-            trading_pair,
-            market_address,
-            integrator: integrator_address,
-            extend_ref,
-            quote_available: 0,
-        });
-        smart_table::add(
-            integrator_fee_stores_map_ref_mut,
-            trading_pair,
-            IntegratorFeeStoreMetadata {
+            OpenOrdersMetadata {
                 market_id,
                 trading_pair,
                 market_address,
-                integrator: integrator_address,
-                integrator_fee_store_address:
-                    object::address_from_constructor_ref(&constructor_ref),
+                user: user_address,
+                open_orders_address,
             }
         );
     }
 
     public entry fun update_market_parameters(
         econia: &signer,
         market_id_option: vector<u64>,
-        pool_fee_rate_bps_option: vector<u8>,
-        taker_fee_rate_bps_option: vector<u8>,
+        pool_fee_rate_option: vector<u16>,
+        protocol_fee_rate_option: vector<u16>,
```

Comment:
```suggestion
        pool_fee_rate_option: Option<u16>,
        protocol_fee_rate_option: Option<u16>,
```

So yeah like we talked about you should be able to just use `Option<T>` with all of these where `T` is a Move primitive and it will serialize it correctly. Nothing changes on the client/SDK side of things, because Options and vectors are serialized the same exact way, since Options are really just vectors of length 1 with helper functions for unwrapping and some/none.

And then you don't have to have the assertions to check that the length is 1, either

Reply:
Addressed in 38f4d3e